### PR TITLE
add new iwconfig parser

### LINKED ISSUE #329: New parser: iwconfig
Hi guys,

What about we add support for `iwconfig <interface>` to extract the current wifi setup? The output looks like this: 

```sh
wlp0s20f3  IEEE 802.11  ESSID:"BLABLABLA"  
          Mode:Managed  Frequency:5.24 GHz  Access Point: 74:13:C2:8B:C6:45   
          Bit Rate=270 Mb/s   Tx-Power=22 dBm   
          Retry short limit:7   RTS thr:off   Fragment thr:off
          Power Management:on
          Link Quality=46/70  Signal level=-64 dBm  
          Rx invalid nwid:0  Rx invalid crypt:0  Rx invalid frag:0
          Tx excessive retries:0  Invalid misc:329   Missed beacon:0
```

### PR DESCRIPTION
**Pull request by vrince**
* [x] introduce new `iwconfig` parser
* [x] add new unit test + test data (simple case)
* [x] implement `_process` (convert int float and all)
* [x] document schema
* [x] add example in docsrting

Closes #329

### CODE REVIEW

**Comment by kellyjonbrazil** [comment]
Sweet - thank you for the contribution! I’ll take a closer look tomorrow. Much appreciated!

File: jc/parsers/iwconfig.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+        List of Dictionaries. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output: List[Dict] = []
+
+    # for backwards compatibility, preset all fields to None
+    wireless_extension_obj: Dict = {
```
Comment (kellyjonbrazil) [inline jc/parsers/iwconfig.py:95]: You can go ahead and make this a blank dictionary. This was just done for backwards compatibility with older versions of the `ifconfig` parser after I rewrote it.

File: tests/test_iwconfig.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        ubuntu_20_10_iwconfig_raw_json= json.loads(f.read())
+
+
+    def test_iwconfig_nodata(self):
+        """
+        Test 'iwconfig' with no data
+        """
+        self.assertEqual(jc.parsers.iwconfig.parse('', quiet=True), [])
+
+    def test_iwconfig_ubuntu_20_04(self):
```
Comment (kellyjonbrazil) [inline tests/test_iwconfig.py:29]: Might add `_raw` to the function name so it doesn't override the other test.

**Comment by vrince** [comment]
@kellyjonbrazil I moved the test fixture to `generic` no clue how generic `iwconfig` is ... but it felt wrong to put it in ubuntu 20.10 arbitrarily. I also added the parser to the parser list so `iwconfig 2> /dev/null | jc --iwconfig -p` is actually working ;)

**Comment by kellyjonbrazil** [comment]
All good - we can put OS-specific test fixtures in their folders if we run into any OS/distro-specific issues later. Looks very clean!

**Pull request merged**